lib/rules/no-for-in-array.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @fileoverview Disallow iterating over an array with a for-in loop
+ * @author Benjamin Lichtman
+ */
+"use strict";
+const ts = require("typescript");
+const util = require("../util");
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/**
+ * @type {import("eslint").Rule.RuleModule}
+ */
+module.exports = {
+    meta: {
+        docs: {
+            description: "Disallow iterating over an array with a for-in loop",
+            category: "Functionality",
+            recommended: false,
+            extraDescription: [util.tslintRule("no-for-in-array")],
+            url: util.metaDocsUrl("no-for-in-array"),
+        },
+        fixable: null,
+        schema: [],
+        type: "problem",
+    },
+
+    create(context) {
+        return {
+            ForInStatement(node) {
+                if (
+                    !context.parserServices ||
+                    !context.parserServices.program
+                ) {
+                    return;
+                }
```

Comment:
TSLint will warn for each rule enabled that requires type information when it's not provided. Should there be some standard way for these to warn in a similar manner? Would that be a separate issue from this PR?

Reply:
https://github.com/bradzacher/eslint-plugin-typescript/blob/master/lib/util.js#L114